### PR TITLE
docs: resource_deps

### PR DESCRIPTION
**Pull request by landism**

### CODE REVIEW

**Comment by maiamcc** [comment]
Lgtm! except... where ARE the resource docs?

**Comment by landism** [comment]
> Lgtm! except... where ARE the resource docs?

`git add` is hard, ok?

File: docs/resource_dependencies.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+2. If you run `tilt up frontend` to select only some of your Tiltfile's resources,
+    that also implicitly selects all of `frontend`'s transitive dependencies.
+
+A resource is "ready" when:
+* For K8s resources: the pod is running and K8s considers all of its containers ready
+* For docker-compose resources: the container is started (NB: Tilt doesn't currently observe docker-compose health checks)
+* For local resources: the command has succeeded at least once
+
+Some other use cases:
+* Define your resource deps such that teammates who work on different subsets of services
```
Comment (maiamcc) [inline docs/resource_dependencies.md]: `tilt up svc1 svc2` looks the long version, not the short version--i bet there's a way to reword this so it's punchier and gets the point across better (e.g. "`tilt up frontend` brings up frontend, the database, and the asset server -- everything someone needs to develop on `frontend`, without needing to explicitly list the deps.)

**Comment by nicks** [comment]
what's the status of this pr? is this ready to be merged, or waiting on something?

**Comment by landism** [comment]
> what's the status of this pr? is this ready to be merged, or waiting on something?

I still need to address Maia's comment and update the caveats now that the flashing green dot bug is fixed.

**Pull request merged**